site/content/how-to/traffic-management/snippets-filters.md

```diff
@@ -0,0 +1,345 @@
+---
+title: "Snippets Filters"
+weight: 800
+toc: true
+docs: "DOCS-000"
+---
+
+Learn how to use Snippets with the `SnippetsFilter` API.
+
+## Overview
+
+Snippets allow Cluster Operators to insert NGINX configuration into different contexts of the
```

Comment:
I think we need to be clear calling them SnippetsFilters, not just Snippets. And also make it clear that these attach to an HTTPRoute rule and apply to that rule specifically.

Reply:
> I think we need to be clear calling them SnippetsFilters, not just Snippets.

I followed Ingress controller's precedence here: https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/

And think that its good to stick with Snippets because it seems like that's already an established NGINX term? At least with the NIC crowd it will strike a familiar bell if we stick with saying Snippets are configured through SnippetsFilters.

> And also make it clear that these attach to an HTTP/GRPCRoute rule and apply to that rule specifically. 

Though, if a SF sets things in the `main`, `http`, or `http.server` contexts, then the SF doesn't apply to that rule specifically right? I was a little confused on this behavior as well.

Reply:
Fair; after re-reading, I think it makes sense to talk about the general idea of snippets, but then describe specifically what a SnippetsFilter is (because in the future we may also have SnippetsPolicy). I'd also rename the doc to `snippets.md`.

> Though, if a SF sets things in the main, http, or http.server contexts, then the SF doesn't apply to that rule specifically right? I was a little confused on this behavior as well.

True, but I think we should make it clear what the intention is. Maybe it can be phrased better. If someone puts something in main, it really should be because it's required for a configuration for a rule. Technically a user could put anything in there, but that's not what it's for.

Reply:
I've added some information on this, please let me know if what I have is correct and/or if there is more stuff I should add.